### PR TITLE
Change request library for python3 compatibility

### LINKED ISSUE #4: Make tracker compatible with Python 3
Relates to openfisca/openfisca-core#685

**Definition of done :**

Make tracker code compatible with Python 3.

**Details:**

Tracker code is not compatible with Python 3. In order to openfisca/openfisca-core#594 it has to be.

### PR DESCRIPTION
**Pull request by Anna-Livia**
Fixes #4 
The request library had no compatibility with python3.
This PR changes it for the [requests](http://docs.python-requests.org/en/master/) library.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -5,14 +5,14 @@
...
-    version="0.2.0",
+    version="0.3.0",
     description="A Tracker of the OpenFisca Web API usage",
     license="http://www.fsf.org/licensing/licenses/agpl-3.0.html",
     author="",
     packages=find_packages(),
     include_package_data=True,
     install_requires=[
-        'unirest == 1.1.7'
+        'requests == 2.19'
```
Comment (sandcha) [inline setup.py]: `requests` seems to be for synchronous HTTP requests while we are looking fo asynchronous requests.
[grequests](https://pypi.org/project/grequests/) library might help us (other libraries exist).

File: tests/test_piwik.py [hunk 2]
```diff
@@ -14,6 +14,8 @@
 
 # Doesn't check that the action was actually tracked, just that it doesn't crash.
 # You can manually check online at TRACKER_URL that the action was indeed tracked.
+# This test never exits. It needs no be manually exited (ctrl + c).
```
Comment (sandcha) [inline tests/test_piwik.py:17]: Even if the tracker shouldn't stop, this feature might be useful for test. 
To do so, we can add this code and call `stop_test(tracker)`:
```
    def stop_test(tracker):
        def stop_timer(tracker):
            return
        tracker.stop_timer = types.MethodType(stop_timer, PiwikTracker)
```

File: openfisca_tracker/piwik.py [hunk 3]
```diff
@@ -34,11 +29,9 @@ def stop_timer(self):
 
     def send(self):
         with self.lock:
-            post(
+            requests.post(
```
Comment (sandcha) [inline openfisca_tracker/piwik.py]: Using [grequests](https://pypi.org/project/grequests/), the syntax looks like this:
```
            rs = (
                grequests.post(self.url, json={"requests": request, "token_auth": self.token_auth})
                for request in self.requests
                )
            grequests.map(rs)
```

**Review by sandcha — CHANGES_REQUESTED** [review]
Update from synchronous HTTP requests to asynchronous requests?

> And stop test?

File: openfisca_tracker/piwik.py [hunk 4]
```diff
@@ -33,17 +29,23 @@ def stop_timer(self):
...
+                json={"requests": self.requests, "token_auth": self.token_auth},
                 )
+            grequests.map([req], exception_handler=exception_handler)
             self.requests = []
 
+
     def track(self, action_url, action_ip=""):
         tracked_request = "?idsite={}&url={}&cip={}&rec=1".format(self.idsite, action_url, action_ip)
+
+        # Lock in case `send()` and `track()` try to access self.requests simultaneously
```
Comment (sandcha) [inline openfisca_tracker/piwik.py]: 👍

File: tests/test_piwik.py [hunk 5]
```diff
@@ -14,6 +14,8 @@
...
+# It might be caused by the timer thread ser in the piwik.py file.
```
Comment (sandcha) [inline tests/test_piwik.py]: What do you mean by `ser`?

File: openfisca_tracker/piwik.py [hunk 6]
```diff
@@ -1,20 +1,16 @@
...
 log = logging.getLogger('gunicorn.error')
 BUFFER_SIZE = 30  # We send the tracked requests by group
 TIMER_INTERVAL = 3600  # We send the tracked requests every TIMER_INTERVAL seconds
 
 
-def default_callback(response):
-    return
-
-
+# Each of openfisca-core instance creates a tracker object (1 per worker).
```
Comment (sandcha) [inline openfisca_tracker/piwik.py]: As I understand it, we have multiple gunicorn sync workers that handle/share only one instance of openfisca web api (= one flask app). This instance has one tracker. 
So, we have only one openfisca-core instance and one tracker instance.

File: openfisca_tracker/piwik.py [hunk 7]
```diff
@@ -33,17 +29,23 @@ def stop_timer(self):
...
                 self.url,
-                headers = {"Accept": "application/json"},
-                params = json.dumps({"requests": self.requests, "token_auth": self.token_auth}),
-                callback = default_callback
+                json={"requests": self.requests, "token_auth": self.token_auth},
                 )
+            grequests.map([req], exception_handler=exception_handler)
             self.requests = []
 
+
```
Comment (sandcha) [inline openfisca_tracker/piwik.py]: Delete this line to make flake8 happy? 🙂

**Review by sandcha — CHANGES_REQUESTED** [review]
Changes on comments (mainly a comment on the number of tracker instances).

File: tests/test_piwik.py (hunk 5 above)
Comment (Anna-Livia) [reply tests/test_piwik.py]: Typo :)

**Review by sandcha — APPROVED** [review]

File: openfisca_tracker/piwik.py (hunk 6 above)
Comment (Anna-Livia) [reply openfisca_tracker/piwik.py]: I traced the tracker object with their ID and `requests`size.
There are 3 distincts tracker object instance when using the API.
The `requests`  length of each increase at different times.

This is why I beleive there are multiple trackers.

**Pull request merged**